web/tests/compose_validate.test.js

```diff
@@ -854,3 +863,75 @@ test_ui("test warn_if_topic_resolved", ({override, mock_template}) => {
     compose_validate.warn_if_topic_resolved(false);
     assert.ok(!error_shown);
 });
+
+test_ui("test_warn_if_guest_in_dm_recipient", ({override_rewire, mock_template}) => {
+    let is_active = false;
+
+    mock_template("compose_banner/guest_in_dm_recipient_warning.hbs", false, (data) => {
+        assert.equal(data.classname, compose_banner.CLASSNAMES.guest_in_dm_recipient_warning);
+        assert.equal(
+            data.banner_text,
+            $t({defaultMessage: "Guest is a guest in this organization."}),
+        );
+        is_active = true;
+        return "<banner-stub>";
+    });
+
+    compose_state.set_message_type("private");
+    initialize_pm_pill(mock_template);
+    compose_state.private_message_recipient("guest@example.com");
+    const classname = compose_banner.CLASSNAMES.guest_in_dm_recipient_warning;
+    let $banner = $(`#compose_banners .${CSS.escape(classname)}`);
+
+    // if setting is disabled, remove warning if exists
+    realm.realm_enable_guest_user_dm_warning = false;
+    compose_validate.warn_if_guest_in_dm_recipient();
+    assert.ok(!is_active);
+
+    // show warning for guest emails, banner should be shown
+    realm.realm_enable_guest_user_dm_warning = true;
+    $banner.length = 0;
+    compose_validate.warn_if_guest_in_dm_recipient();
+    assert.ok(is_active);
+    assert.equal(
+        compose_state.get_recipient_guest_emails_for_dm_warning().join(","),
+        "guest@example.com",
+    );
```

Comment:
In a test, if we expect only 1 itemed array in `get_recipient_guest_emails_for_dm_warning`, then we can assert the size of the array + access the 1st element directly. Having that join there confused me when comparing to a string with no commas.